Tutorial - Vue feature flags (#7555)
PostHog/posthog.com: contents/tutorials/vue-feature-flags.md

## Implementing our flag code

Once created, we can add our feature flag to our app. We do this using the `this.$posthog.onFeatureFlags` callback.

In our example, we fetch our feature flag and update the text on the page based on its value:

```vue file=App.vue
<template>
  <div id="app">
    <h1>{{ titleText }}</h1>
  </div>
</template>

<script>
export default {
  name: 'App',
  data() {
    return {
      titleText: 'Waiting for feature flag'
    };
  },
  mounted() {
    const posthog = this.$posthog;
    posthog.onFeatureFlags(() => {
      if (posthog.isFeatureEnabled('my-cool-flag')) {
        this.titleText = 'Our flag is enabled!';
      } else {
        this.titleText = 'Our flag is disabled!';
      }
    });
  },
}
</script>
```

When you run your app now, you should see the updated text.

![New app after adding the flag](../images/tutorials/vue-feature-flags/flag-enabled-in-app.png)

### Bootstrapping feature flags

> You may notice flickering when you load your app. This is because PostHog is making an asynchronous request to fetch the flag value and so the text is only updated when the request returns. To have your feature flags available immediately, you can initialize PostHog with precomputed values until PostHog has had a chance to fetch them. This is called [bootstrapping](/docs/feature-flags/bootstrapping).

Reviewer: I think this paragraph could be simplied, one sentence less probably. 
```suggestion
You may notice the text flickers when you load your app. This is because PostHog is making an asynchronous request to fetch the flag value and the text only updates when the request returns. To prevent this and have your feature flags available immediately, you can initialize PostHog with precomputed flag values until it has a chance to fetch them. This is called [bootstrapping](/docs/feature-flags/bootstrapping).
```

To do this, use the `bootstrap` key in PostHog's initialization config and add feature flag values to it:

```js file=plugins/posthog.js
import posthog from "posthog-js";

export default {
  install(app) {
    app.config.globalProperties.$posthog = posthog.init(
      "<ph_project_api_key>",
      {
        api_host: "<ph_instance_address>",
        bootstrap: {
            featureFlags: {
                'my-cool-flag': true,
            },
        },
      }
    );
  },
};
```

> Now when you reload your app, the flicker is gone. See our [docs](/docs/feature-flags/bootstrapping) for more information on bootstrapping best practices.

Reviewer: This is a good way of showing bootstrapping, but you might want to add a note about how the server can get this value before the client loads. Gives them a hint of what to do to make this work properly.

> Now when you reload your app, the flicker is gone. See our [docs](/docs/feature-flags/bootstrapping) for more information on bootstrapping best practices.

Author: ```suggestion
Now when you reload your app, the flicker is gone. 

To bootstrap PostHog with more accurate flag values, you can return the initial values during your app's first page load request. See our [docs](/docs/feature-flags/bootstrapping) for more information on how to do this.
```
